iothub_client/src/iothub_client_ll_uploadtoblob_uhttp.c

```diff
@@ -740,7 +743,13 @@ static int initiate_blob_upload(IOTHUB_CLIENT_LL_UPLOADTOBLOB_HANDLE_DATA* uploa
 
                         if ((hub_content = STRING_construct(FILE_UPLOAD_ABORTED_BODY)) != NULL)
                         {
-                            if (IoTHubClient_LL_UploadToBlob_step3(upload_data, STRING_c_str(correlation_id), http_client, request_header, hub_content) != 0)
+                            if ((http_client = create_http_client(upload_data)) == NULL)
```

Comment:
I'd like to get some idea f2f on what was the reasoning for adding create_http_client here and other places.

Reply:
Sure: this is due to how uhttp handles xio when a proxy is configured. 
@jebrando may have more context as he was the original designer of the BLOB over uhttp implementation.